src/ArcxAnalyticsSdk.ts

```diff
@@ -75,6 +95,80 @@ export class ArcxAnalyticsSdk {
     )
   }
 
+  private async _onAccountsChanged(accounts: string[]) {
+    if (!window.ethereum) {
+      throw new Error('ArcxAnalyticsSdk::_onAccountsChanged: No ethereum provider found')
+    }
+
+    if (accounts.length > 0) {
+      if (accounts[0] === this.previousConnectedAccount) {
+        // We have already reported this account
+        return
+      } else {
+        this.previousConnectedAccount = accounts[0]
+      }
+
+      const chainIdHex = await window.ethereum.request<string>({ method: 'eth_chainId' })
+      // Because we're connected, the chainId cannot be null
+      if (!chainIdHex) {
+        throw new Error('ArcxAnalyticsSdk::_onAccountsChanged: chainIdHex is:' + chainIdHex)
+      }
+      this.previousChainId = parseInt(chainIdHex, 16).toString()
```

Comment:
Why do we need to parse hexadecimal chain id?

Reply:
should `_onAccountsChanged` modify chain id? it should handle only account

Reply:
The chainId comes as hexadecimal, and the pipelines don't handle hex-to-int transformation. Instead of reporting a hex, we might as well transform the chain id here.

Reply:
It think it should handle the chain ID because when a user connects or disconnects, the current chain ID should be changed

Reply:
It shouldn't from my perspective. If chain id changes as well, then `CHAIN_CHANGED` should catch it.

Reply:
> he chainId comes as hexadecimal, and the pipelines don't handle hex-to-int transformation. Instead of reporting a hex, we might as well transform the chain id here.

What if customer get this as hexadecimal and put it like this?